public/app/types/alerting.ts

```diff
@@ -144,7 +144,7 @@ export interface NotificationChannelOption {
   validationRule: string;
   subformOptions?: NotificationChannelOption[];
   dependsOn: string;
-  setValueAs?: (value: string | boolean) => string | number | boolean | null;
+  setValueAs?: (value: string | boolean) => string | number | boolean | null | undefined;
```

Comment:
why do we need to add this change?

Reply:
I tried a different approach here first with `setValueAs()` returning `undefined` for validation but `validationRule` worked better here, should revert this.